tests/func/test_data_status.py

```diff
@@ -176,10 +176,32 @@ def test_skip_uncached_pipeline_outputs(tmp_dir, dvc, run_copy_metrics):
     )
 
 
-def test_output_with_newly_added_stage(tmp_dir, dvc):
+def test_outs_with_no_hashes(M, tmp_dir, dvc, scm):
+    dvc.stage.add(single_stage=True, outs=["bar"])
     dvc.stage.add(deps=["bar"], outs=["foo"], name="copy", cmd="cp foo bar")
-    assert dvc.data_status() == EMPTY_STATUS
-    assert dvc.data_status(granular=True) == EMPTY_STATUS
+
+    expected_output = {
+        **EMPTY_STATUS,
+        "committed": {"added": M.unordered("bar", "foo")},
```

Comment:
Since there are no hashes, `committed` should not return any result, which was the previous implementation.

This is a bug in `dvc_data.index.diff()` implementation.

Reply:
@efiop, what do you think about the above issue? Should this be fixed in upstream or should be handled in `data status`? Let me know what you think.

Reply:
@skshetry Not sure I totally follow. So `committed` is head vs workspace, and head has nothing and workspace has stuff (foo, bar) which means that status is `added` for those. Or am I missing something?

Reply:
No, `committed` is between head and index. head has nothing, index has a .dvc file without any md5s.

We used to ignore them, since there are no hashes. But now it returns as added.

Reply:
Also i understand this scenario is artificial. And any decision we make here does not matter that much. Just wanted to let you know that it deviates how we treated stuff without hashes.

Reply:
@skshetry Sorry, I meant `head` and `index`. But yeah, since we are comparing hashes only, it is this strange behavior that we need to clarify. It does make sense to have it in `dvc-data`'s `index.diff` indeed. I'll take a look.

Reply:
For the record https://github.com/iterative/dvc/pull/8885